### DISCUSSION TITLE
How can I skip empty x axis with scaleTime?

### DISCUSSION THREAD

**Discussion by gregg-cbs** [f5/unovis#498] [Q&A]
My xScale is Scale.scaleTime() which looks like its going up in a linear fashion even when i dont have data for certain days.

My data does not have entries for the 3rd, 6th, 9th and 15th November but bar columns are being created for those days.

How do i not create bars for days that are not in my data?

Screenshot:
![image](https://github.com/user-attachments/assets/d576f7fe-06a2-489f-9ebf-e347266558f0)

**Comment by lee00678** [comment]
@gregg-cbs one quick solution I can think of is to turn your xAxis into an ordinal scale instead of linear. See this: https://unovis.dev/docs/guides/tips-and-tricks/#displaying-ordinal-values

**Reply by gregg-cbs** [reply]
Thanks for the response, it tried this today but i kept getting an error.

When I use the ordinal scale I get an error:
"Uncaught TypeError: axisGen.scale().ticks is not a function"

```svelte
const xScale = Scale.scaleOrdinal()
const tickFormatX = (tick: number) => data[tick].x_label

    <VisAxis 
      type="x" 
      numTicks={Math.min(data.length, 10)}
      gridLine={true} 
      tickLine={false} 
      tickTextWidth={20}
      tickFormat={tickFormatX}
    />
```

I cant find any info in the docs on how to use or setup an ordinal scale.

**Reply by rokotyan** [reply]
@gregg-cbs Check out this page, may be it'll help https://unovis.dev/docs/guides/tips-and-tricks#displaying-ordinal-values
